### ISSUE TITLE
[BUG] Delete an image just Actived may cause data out-of-sync between Harvester and Longhorn

### ISSUE THREAD

**Issue by albinsun** [harvester/harvester#4027]
## Describe the bug
Delete an Active image which is just finish uploading may cause dirty data on LH since it's still `Importing`. This may affect following operation or test.

## To Reproduce
1. Go to 'Images'
2. Click 'Create' on the upper right corner
3. Upload by file (then will automatically return to Images page)
4. Image sttus should be `Uploading`
5. When status becomes `Active`, delete it immediately.
7. Image is deleted on Harvester, but still extsted on LH.

Video example:
<details>
 <summary>Manual Test (Video)</summary>

 https://github.com/harvester/harvester/assets/2773781/dee72be1-d5d0-495d-a560-08f7e1701d04
</details>
<details>
 <summary>Auto Test (Video)</summary>
   
 https://github.com/harvester/harvester/assets/2773781/afc16879-5d5b-49e3-8116-7ae2e2a237fd
</details>

## Expected behavior
1. Images status should be the same on both sides (Harvester and Longhorn)
2. Personally I will expect user can proceed further operation on an Active image.

## Support bundle
[supportbundle-deleteWhenActived.zip](https://github.com/harvester/harvester/files/11632573/supportbundle-deleteWhenActived.zip)

## Environment
 - Harvester ISO version: v1.2.0-rc1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):
   * KVM, single node
   * ui-source: Auto
   * Longhorn version: v1.4.2

## Additional context
1. In auto test, currently we use image get API and check by return code 200. The endpoint is `apis/{{API_VERSION}}/namespaces/{ns}/virtualmachineimages/{uid}`
1. Example get API response for `Imported` and `Importing` (both images are `Active` in Images)
   
   ![image](https://github.com/harvester/harvester/assets/2773781/1ac70d88-aaae-44fa-bfee-57ad3c71516c)

**Comment by khushboo-rancher** [comment]
Logs from longhorn-manager
```
[longhorn-manager-kfzjd longhorn-manager] time="2023-06-02T22:03:40Z" level=warning msg="Waiting until backing image data source is cleaned before removing the finalizer" backingImageName=default-07h00m38s754888-05-30 controller=longhorn-backing-image node=harvester-d4csq
[longhorn-manager-kfzjd longhorn-manager] time="2023-06-02T22:03:40Z" level=warning msg="Error syncing Longhorn backing image longhorn-system/default-07h00m38s754888-05-30" error="failed to sync backing image for longhorn-system/default-07h00m38s754888-05-30: backing image data source status is ready not failed-and-cleanup"
```

**Comment by lanfon72** [comment]
Update labels, this can be always reproduced in our automation testing, all scenarios of `recreate_image` will encountering this.

**Comment by guangbochen** [comment]
will need to bump LH to v1.4.3 and verify it, thanks.

**Comment by WebberHuang1118** [comment]
This ticket contains two issues:
1. First one is duplicated with #1947, the virtual image deletion got stuck if it's deleted just after the uploading starts. This happens since Longhron v1.4.2, because there is a new scenario to handle backing images. please refer to https://github.com/harvester/harvester/issues/1947#issuecomment-1578179066 for more details
2. Second one is a corner case for virtual image deletion, it happens if the virtual image is deleted just after the file upload completes but before the virtual image gets ready, it should be resolved by Longhorn ticket https://github.com/longhorn/longhorn/issues/6086

**Comment by guangbochen** [comment]
This is covered in the LH [v1.4.3-rc1](https://github.com/longhorn/longhorn/releases/tag/v1.4.3-rc1) release, please help to verify it first using the rc version, thanks.

**Comment by WebberHuang1118** [comment]
Verified with Longhorn v1.4.3-rc1, from pure Longhorn environment and Harvester. This issue is fixed in LH [v1.4.3-rc1](https://github.com/longhorn/longhorn/releases/tag/v1.4.3-rc1) release.

Verification steps on pure Longhorn environment
- Build k8s cluster from scratch
- Install Longhorn v1.4.3-rc1
- Following steps in https://github.com/longhorn/longhorn/issues/6086#issue-1747197062
- Result: pass

Verification steps on Harvester
- Creating Harvester with ISO built from https://github.com/WebberHuang1118/harvester/tree/longhorn-1.4.3-rc3
- Following steps in https://github.com/harvester/harvester/issues/4027#issue-1737600918
- Result: pass

**Comment by albinsun** [comment]
Tested in `v1.2.0-rc4`, close as fixed.
<details>
   <summary>Upload and delete immediatly when Active for 3 times :heavy_check_mark:</summary> 

   https://github.com/harvester/harvester/assets/2773781/612655c4-7ed7-462c-9dbd-a78e79ed388e
</details>

Note that automation TC hits an extended issue when next upload during previous deletion still in-progress, trace in [harvester#4293](https://github.com/harvester/harvester/issues/4293).
